File: backend/src/main/java/io/chesslave/app/EventServlet.java
package io.chesslave.app;

import org.eclipse.jetty.websocket.servlet.WebSocketCreator;
import org.eclipse.jetty.websocket.servlet.WebSocketServlet;
import org.eclipse.jetty.websocket.servlet.WebSocketServletFactory;

public class EventServlet extends WebSocketServlet {

    private final WebSocketCreator creator;

    public EventServlet(WebSocketCreator creator) {
        this.creator = creator;
    }

    @Override
    public void configure(WebSocketServletFactory factory) {
        factory.setCreator(creator);
    }
}

File: backend/src/main/java/io/chesslave/app/EventServlet.kt
package io.chesslave.app

import org.eclipse.jetty.websocket.servlet.WebSocketCreator
import org.eclipse.jetty.websocket.servlet.WebSocketServlet
import org.eclipse.jetty.websocket.servlet.WebSocketServletFactory

class EventServlet(val creator: WebSocketCreator) : WebSocketServlet() {

    override fun configure(factory: WebSocketServletFactory) {
        factory.creator = creator
    }
}
